fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1768,12 +1797,56 @@ public TScanRangeLocation selectBackendsByRoundRobin(List<TScanRangeLocation> lo
         return location;
     }
 
+    private void computeScanRangeAssignmentByConsistentHash(
+            ScanNode scanNode,
+            final List<TScanRangeLocations> locations,
+            FragmentScanRangeAssignment assignment,
+            Map<TNetworkAddress, Long> assignedBytesPerHost,
+            Map<TNetworkAddress, Long> replicaNumPerHost) throws Exception {
+        Collection<Backend> aliveBEs = idToBackend.values().stream().filter(SimpleScheduler::isAvailable)
+                .collect(Collectors.toList());
+        if (aliveBEs.isEmpty()) {
+            throw new UserException("No available backends");
+        }
+        int virtualNumber = Math.max(Math.min(512 / aliveBEs.size(), 32), 2);
+        ConsistentHash<TScanRangeLocations, Backend> consistentHash = new ConsistentHash<>(
```

Comment:
TODO: I think we can move this ConsistentHash to `SystemInfoService`.
No need to create one for every query